### DISCUSSION TITLE
Clipboard access issue

### DISCUSSION THREAD

**Discussion by chris-fj** [CollaboraOnline/online#9038] [Q&A] [answered]
Hi everyone,

I have Collabora Online with this version:
```
COOLWSD version: 
24.04.1.4
git hash: 
 [f12ee1f]
```

on top of ownCloud 10.14.0 (stable) and behind Traefik to provide SSL and the whole stack runs in Docker. Everything seems to work well, except the fact that the clipboard doesn't work, because when I open the developer tools (when trying to copy something), I get this error:

```
Permissions policy violation: The Clipboard API has been blocked because of a permissions policy applied to the current document
```

I assume there's something still to be configured in Collabora to prevent this error from happening, but I haven't been able to find it. Can someone help me?

**Comment by vmiklos** [comment]
https://sdk.collaboraonline.com/docs/advanced_integration.html#allow-the-clipboard-permission-query

Could you please check if your iframe has these magic attributes to allow clipboard access? If not, that needs adding, probably. I guess that needs filing a github issue against https://github.com/owncloud/richdocuments

**Reply by chris-fj** [reply]
Hey @vmiklos, thanks for your quick reply.

I opened a document (spreadsheet), opened the developer tools and inspected the elements. On the resulting HTML code, I searched for `iframe` and I found three instances, none of which have the properties in the linked document. The three `iframe` elements are shown below in the order they appear:

```html
<iframe id="loleafletframe" name="loleafletframe" allowfullscreen="" style="width:100%;height:100%;position:absolute;" onload="this.contentWindow.focus()"></iframe>
...
<iframe class="resize-detector" title="Intentionally blank" aria-hidden="true"></iframe>
...
<iframe class="" style="display: none;"></iframe>
```

I think the most important is the first one. Should I open this issue in the tracker you specified? 

Thanks

**Reply by vmiklos** [reply, marked as answer]
Yes, please do so. Feel free to link the opened issue here, so it also gets some attention from the CollaboraOnline side. Thanks.

**Reply by krzys-h** [reply]
He did open the issue ;)

https://github.com/owncloud/richdocuments/issues/545

**Reply by chris-fj** [reply]
Oh yeah, I opened the issue but forgot to cross post it here! Thanks for bringing it :)
